utils/s2n_random.c

```diff
@@ -332,8 +341,84 @@ S2N_RESULT s2n_get_private_random_bytes_used(uint64_t *bytes_used)
     return S2N_RESULT_OK;
 }
 
-static int s2n_rand_urandom_impl(void *ptr, uint32_t size)
+S2N_RESULT s2n_rand_get_dev_urandom(struct s2n_rand_device **device)
+{
+    RESULT_ENSURE_REF(device);
+    RESULT_ENSURE(s2n_in_unit_test(), S2N_ERR_NOT_IN_UNIT_TEST);
+    *device = &s2n_dev_urandom;
+    return S2N_RESULT_OK;
+}
+
+S2N_RESULT s2n_rand_device_open(struct s2n_rand_device *device)
+{
+    RESULT_ENSURE_REF(device);
+    RESULT_ENSURE_REF(device->source);
+
+    DEFER_CLEANUP(int fd = -1, s2n_rand_entropy_fd_close_ptr);
+    while (fd < 0) {
+        fd = open(device->source, ENTROPY_FLAGS);
+        if (fd < 0 && errno != EINTR) {
+            RESULT_BAIL(S2N_ERR_OPEN_RANDOM);
+        }
```

Comment:
Can you use https://github.com/aws/s2n-tls/blob/main/utils/s2n_io.h#L22? The EINTR loops can be kind of hard to read when combined with normal error handling.

Reply:
I forgot about this, thanks!